gobblin-utility/src/main/java/org/apache/gobblin/util/HadoopUtils.java

```diff
@@ -757,6 +761,57 @@ public static void safeRenameRecursively(FileSystem fileSystem, Path from, Path
     }
   }
 
+  public static void ensureDirectoryExists(FileSystem fs, Path path, Iterator<OwnerAndPermission> ownerAndPermissionIterator)
+      throws IOException {
+
+    if (fs.exists(path)) {
+      return;
+    }
+
+    if (ownerAndPermissionIterator.hasNext()) {
```

Comment:
the `else` case is so brief (and of paramount importance as the base in the recursion) while the `if` case so long, that I'd suggest inverting the test to put that first.  i.e.:
```
if (fs.exists(path)) {
  return;
} else if (!ownerAndPermissionIterator.hasNext()) {
  fs.mkdirs(path);
} else {
  OwnerAndPermission ...
}
```